### DISCUSSION TITLE
What is the best approach to test bots?

### DISCUSSION THREAD

**Discussion by elebur** [python-telegram-bot/python-telegram-bot#3639] [Q&A]
I wrote a bot, and now I want to test its handlers to ensure that everything works like I want. How can I do it?

Are there any ready-made modules for this purpose?

**Comment by njbbaer** [comment]
I would also like to know this! I want to test my bot with pytest.

**Comment by harshil21** [comment]
There isn't. There used to be https://github.com/python-telegram-bot/ptbtest but that is very outdated. We are accepting maintainers for that provided you can demonstrate familiarity and experience in python and ptbtest. (https://t.me/pythontelegrambotchannel/115)
